servers/rendering/renderer_rd/forward_clustered/scene_shader_forward_clustered.cpp

```diff
@@ -829,6 +836,14 @@ void SceneShaderForwardClustered::init(const String p_defines) {
 
 		actions.usage_defines["MODEL_MATRIX"] = "#define MODEL_MATRIX_USED\n";
 
+		String point_size_usage_defines = "#define POINT_SIZE_USED\n";
+		if (emulate_point_size) {
+			point_size_usage_defines += "#define EMULATE_POINT_SIZE\n";
+		}
```

Comment:
This will break shader baking. Since if you bake a shader when the editor is running Vulkan, it will report that point size is supported. But then you will end up with a D3D12 shader without emulate point size being true. 

In https://github.com/godotengine/godot/pull/102552 Dario had to systematically remove all the places where we changed shader code based on the underlying platform. 

Off the top of my head I don't see an easy way out of this issue unless we can somehow push these changes down to an even lower level.